fix: Check for equality of value instead of equality of instance.

michael-simons opened the pull request:
isCustomized returned the wrong value for all instances that have been serialized and read back as they obviously does not point to the default instance.

The solution is to check for equality of the values, not for equality of the instances. This is a minor change in behaviour: A setting that is created with the builder matching the exact same defaults is now considered to be not customized.
This solution has been preferred over a custom readResolve returning the DEFAULT when things match as it would otherwise be able to create an instance that identifies itself as customized but changed to not customized when read back, basically the oppositte of what is fixed here: An instances created as not customized must never be read back as customized.

michael-simons committed:
fix: Check for equality of value instead of equality of instance.

isCustomized returned the wrong value for all instances that have been serialized and read back as they obviously does not point to the default instance.

The solution is to check for equality of the values, not for equality of the instances. This is a minor change in behaviour: A setting that is created with the builder matching the exact same defaults is now considered to be not customized.
This solution has been preferred over a custom readResolve returning the DEFAULT when things match as it would otherwise be able to create an instance that identifies itself as customized but changed to not customized when read back, basically the oppositte of what is fixed here: An instances created as not customized must never be read back as customized.

mattcasters commented on driver/src/main/java/org/neo4j/driver/internal/SecuritySettings.java:
```diff
@@ -59,7 +60,20 @@ public boolean encrypted()
     private boolean isCustomized()
     {
-        return this != DEFAULT;
+        return !(DEFAULT.encrypted() == this.encrypted() && DEFAULT.hasEqualTrustStrategy( this ));
```
Indeed this is the source of contention across serialization exercises. This change would fix that.

michael-simons committed:
Adapt test.

These tests needed to be adapted due to the fact that the previous test did not actually change the default value.
